hack/lib/util.sh

```diff
@@ -139,6 +135,23 @@ kube::util::gen-doc() {
   ${cmd} "${tmpdir}"
   # create the list of generated files
   ls "${tmpdir}" | LC_ALL=C sort > "${tmpdir}/.files_generated"
+
+  # remove all old generated file from the destination
+  for file in $(cat "${dest}/.files_generated" 2>/dev/null); do
+    if [[ -e "${tmpdir}/${file}" && ! -z "${skipline}" ]]; then
+      # If RCS format diff has 3 lines, only 1 line changed.
+      local fdiff=$(diff -n "${dest}/${file}" "${tmpdir}/${file}")
+      if [[ $(echo "${fdiff}" | wc -l) == 3 && $(echo "${fdiff}" | grep -e "${skipline}") ]]; then
```

Comment:
In which case, may I suggest filtering out lines starting with `"^###### Auto generated by spf13/cobra at"` and then comparing that the contents are exactly the same? Or instead of filtering them out, replacing them with something like `"###### Auto generated by spf13/cobra at XXXX-XX-XX XX:XX:XX.XXXXXXXXX +XXXX XXX"`.

If your use case is fairly specific, I think that's a lot superior than trying to interpret diff output from bash code... Maybe replace "skipline" with "prefix" and assume you know what the regexp for a timestamp you recognize is or should be?